src/WorldStorage/WSSAnvil.cpp

```diff
@@ -1343,7 +1343,7 @@ OwnedBlockEntity cWSSAnvil::LoadMobSpawnerFromNBT(const cParsedNBT & a_NBT, int
 	if ((Type >= 0) && (a_NBT.GetType(Type) == TAG_Short))
 	{
 		short MonsterType = a_NBT.GetShort(Type);
-		if ((MonsterType >= 50) && (MonsterType <= 120))
+		if (MonsterType != eMonsterType::mtInvalidType)
 		{
 			MobSpawner->SetEntity(static_cast<eMonsterType>(MonsterType));
```

Comment:
I'm not sure about this piece of code anymore. Since monster types no longer have a fixed value, this will probably break if we add new mobs. https://github.com/cuberite/cuberite/commit/01b8ed5295875262a91b60af878bf2a18c1b7aae#diff-c156838c409845aa17aefa4013707a49

Reply:
Why do we have this special cause for Cuberite worlds to begin with?

Reply:
A quick note:

When I began with this PR the spawner weren't loaded properly.
I wondered why this checked for 50->120 so I removed it.

We could just dump the numerical saving and use the Minecraft:namespace way. This only should have a minor impact on startup/shutdown and the extra disc space doesn't matter.

Reply:
The thing breaking the whole thing is that if the check ```if ((MonsterType >= 50) && (MonsterType <= 120))``` fails the check for the string isn't even tried. We could also change it that way that we check for string anyway

Reply:
I'm in favor of dropping the numerical saving and just sticking with the vanilla variant.

Reply:
All in favour raise you hand 🙋🏻‍♂️

Reply:
Side note: we already write both (string and short) into the save file so no problems on compability

Reply:
Make sure to add the relevant dataversion 2566 (1.16.2 I think) eMobType <-> namespaced IDs mapping to WorldStorage\NamespaceSerializer

Reply:
The Id conversion is done by ````cMonster::MobTypeToVanillaName```. Do we move this there?

Reply:
I had a look at the code doing the conversion... it's just iterating through a list of all availeble entities and returning the one matching - realy nice :D

New PR or is it ok to do this together?

Reply:
I am gonna do the back and forth conversion in another PR: There is some more stuff to do:
 - Move the mob enum to registries
 - add the matching functions to NamespaceSerializer
 - ...

Reply:
I did some research...

1.12 and earlier doesn't load the right entity because the name is written in ThisWay
1.13 doesn't even parse chunks...

Reply:
Hopefully once Anvil storage is fully upgraded we can leapfrog directly to full compatibility with 1.16 :)

Reply:
@mathiascode you are right. If we add another mob (and there many left) in alphabetical order we don't keep the same numerical IDs so the number idea is VERY bad as this is dependent of the order in the enum